controllers/extendeddaemonset/controller.go

```diff
@@ -262,9 +262,11 @@ func (r *Reconciler) updateInstanceWithCurrentRS(logger logr.Logger, daemonset *
 
 	// Check if newDaemonset differs from existing daemonset, and update if so
 	if !apiequality.Semantic.DeepEqual(daemonset, newDaemonset) {
+
 		if updateDaemonsetSpec {
-			if err := r.client.Update(context.TODO(), newDaemonset); err != nil {
-				return newDaemonset, reconcile.Result{}, err
+			newDaemonsetCopy := newDaemonset.DeepCopy()
```

Comment:
I'm not sure why this DeepCopy() call seems to have been removed in #39 . Without it both the spec and the status do not update properly; only the one called first gets updated.

Reply:
this is now resolved